Add Deployment Guide and update links

roooms opened the pull request.

pearkes commented on website/source/docs/guides/deployment.html.md:
```diff
@@ -1,12 +1,13 @@
 ---
 
-# Consul Production Deployment Guide
+# Consul Reference Architecture
```
Just wanted to note that I'm not 100% sure "Reference Architecture" is the right phrasing here for this guide. We may want to consider other options. Once we finalize a name we should also align the names of the guides with the filenames, currently having two at `deployment*.html` may be kind of confusing to users / search engines.

roooms commented:
Thanks for the feedback, Jack.

You’re right that the current version of this document is somewhat of a
hybrid between a reference architecture and a deployment guide. It covers
hardware sizing and network requirements but also configuration
recommendations like raft multiplier settings.

It was intended to be a reference architecture but was developed before we
clarified the scope of each document type.

With regards to naming, I created the new deployment guide in this PR but
didn’t want to break the old URL. The intention was a future change would
rename the reference architecture file to be more descriptive and introduce
the necessary config to handle a redirect so we don’t return a 404 on the
old name. We can include that in this PR, it’s just I didn’t know how the
redirect is handled.


On Fri, 3 Aug 2018 at 22:49, Jack Pearkes <notifications@github.com> wrote:

> *@pearkes* commented on this pull request.
>
> Haven't been through this in detail but wanted to ask about naming.
> ------------------------------
>
> In website/source/docs/guides/deployment.html.md
> <https://github.com/hashicorp/consul/pull/4487#discussion_r207675410>:
>
> >  ---
>
> -# Consul Production Deployment Guide
> +# Consul Reference Architecture
>
> Just wanted to note that I'm not 100% sure "Reference Architecture" is the
> right phrasing here for this guide. We may want to consider other options.
> Once we finalize a name we should also align the names of the guides with
> the filenames, currently having two at deployment*.html may be kind of
> confusing to users / search engines.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/hashicorp/consul/pull/4487#pullrequestreview-143344557>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEF0WgHBfIKiqa_d1xu5flrpS2-XMA3Qks5uNMVSgaJpZM4Vugv_>
> .
>
-- 
Dan Brown
Director, Enterprise Architecture
dan@hashicorp.com
www.hashicorp.com

roooms commented:
@topfunky @pearkes I have added some more commits to this PR to get it ready for merging:

- Cut down "deployment" doc (which should focus on Reference Architecture) by moving raft and performance tuning to the Server Performance page which already covers some of this.
- Moved backups from "deployment" doc (which should focus on Reference Architecture) to "deployment-guide".

The end result is the _Production Deployment_ doc is renamed _Reference Architecture_ to match other products reference architecture sections and the _Deployment Guide_ can be published to cover the installation and configuration of Consul. 

I have not renamed the deployment.html.md doc to reference-architecture.html.md in this series of commits, as we can do this in a separate pass when the navigation is restructured (based on the internal review we conducted) and redirects are created for URL changes.
